src/components/Single-product-detail/SingleProductDetails.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import "./singleProductDetails.css";
+import { useParams } from "react-router-dom";
+import { Carousel } from "../Carousel/Carousel";
+import { useCart } from "../../contexts/cart-context";
+import { useWishlist } from "../../contexts/wishlist-context";
+import { getOriginalPrice } from "../../utils/productutilFuncs";
+import { useProductsDetail } from "../../contexts/productsDetail-context";
+
+const SingleProductDetails = () => {
+
+    const { productId } = useParams();
+    const { productsDataFetchedFromApi: productsData } = useProductsDetail();
+    const { stateOfWishlist, dipatchOfWishlist } = useWishlist();
+    const { stateOfCart, dispatchOfCart } = useCart();
+
+    const getTheProductDetails = (productsData, productId) => {
+         return productsData.find(product => product._id === productId)
+    }
+
+    const singleProduct = getTheProductDetails(productsData, productId)
+
+    return (
+        <div className="ap_single-product-detail-component">
+            <div className="ap_single-product-container">
+                <div className="ap_singleProduct-carousel-container">
+                    <Carousel>
+                        {
+                            singleProduct.img.map((productImg, index) => (
+                                <div className="ap_single-product-img" key={index}>
+                                    <img src={ productImg } />
+                                </div>
+                            ))
+                        }
+                    </Carousel>
+                </div>
+                <div className="ap_singleProduct-details-container a-tl">
+                    <h3 className="ap_singleProduct-title">{ singleProduct.brandName }</h3>
+                    <p className="ap_singleProduct-description">{ singleProduct.description }</p>
+                    { singleProduct.isNewArrival? <div className="ap_singleProduct-is-newArrival">New Arrival</div> : "" }
+                    { singleProduct.isTrending? <div className="ap_singleProduct-is-trending">Trending</div> : "" }
+                    { !singleProduct.isInStock? <div className="ap_singleProduct-is-stockOut">Stock Out</div> : "" }
```

Comment:
Instead of the ternary operator `?` you can use the logical AND operator `&&` to achieve the same.
```jsx
{ singleProduct.isNewArrival && <div className="ap_singleProduct-is-newArrival">New Arrival</div> }
```

Reply:
thanks for point that out, didn't noticed it!!